Content.Server/Box/BoxSystem.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Linq;
+using Content.Server.Box.Components;
+using Content.Server.Storage.Components;
+using Content.Shared.Box;
+using Content.Shared.Box.Components;
+using Content.Shared.Movement.Components;
+using Robust.Shared.Player;
+
+namespace Content.Server.Box;
+
+public sealed class BoxSystem : SharedBoxSystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<BoxComponent, StorageBeforeCloseEvent>(OnBeforeStorageClosed);
+        SubscribeLocalEvent<BoxComponent, StorageAfterOpenEvent>(AfterStorageOpen);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var box in EntityQuery<BoxComponent>())
+        {
+            if (box.EffectPlayed)
+                box.Accumulator += frameTime;
+
+            if (box.Accumulator >= box.EffectCooldown)
+            {
+                box.Accumulator -= box.EffectCooldown;
+                box.EffectPlayed = false;
+            }
+        }
```

Comment:
Same logic that applied to the chameleon stuff applies here. Instead of updating an accumulator every tick just have a `TimeSpan? TimeUntilEffectCanHappenAgainOrWhatever`. Fortunately, unlike the chameleon stuff the logic should be much simpler, and should basically involve replacing `EffectPlayed  => _timing,CurTime < TimeUntilEffectCanHappenAgainOrWhatever`